tests/langchain/test_langchain_autolog.py

```diff
@@ -0,0 +1,484 @@
+        with mlflow.start_run() as run:
+            model.invoke("MLflow")
+        client = MlflowClient()
+        artifacts = client.list_artifacts(run.info.run_id)
+        artifacts = [x.path.split(os.sep)[-1] for x in artifacts]
+        for artifact_name in get_mlflow_callback_artifacts(contains_llm=True):
+            if isinstance(artifact_name, str):
+                assert artifact_name in artifacts
+            else:
+                assert any(artifact_name.match(artifact) for artifact in artifacts)
+
+
+def test_loaded_llmchain_autolog():
+    mlflow.langchain.autolog(log_models=True, log_input_examples=True)
+    model = create_openai_llmchain()
+    question = {"product": "MLflow"}
+    answer = {"product": "MLflow", "text": TEST_CONTENT}
+    with _mock_request(return_value=_mock_chat_completion_response()):
+        with mlflow.start_run() as run:
+            assert model.invoke(question) == answer
+        with mock.patch("mlflow.langchain.log_model") as log_model_mock:
+            loaded_model = mlflow.langchain.load_model(f"runs:/{run.info.run_id}/model")
+            assert loaded_model.invoke(question) == answer
+            log_model_mock.assert_not_called()
+
+            mlflow_model = get_mlflow_model(run.info.artifact_uri)
+            model_path = os.path.join(run.info.artifact_uri, MODEL_DIR)
+            input_example = _read_example(mlflow_model, model_path)
+            assert input_example.to_dict("records") == [question]
+
+            pyfunc_model = mlflow.pyfunc.load_model(f"runs:/{run.info.run_id}/model")
+            # _TestLangChainWrapper mocks this result
+            assert pyfunc_model.predict(question) == [TEST_CONTENT]
+            log_model_mock.assert_not_called()
+
+            signature = mlflow_model.signature
+            assert signature == infer_signature(question, [TEST_CONTENT])
+
+
+def test_llmchain_autolog_log_inference_history():
+    mlflow.langchain.autolog(log_inference_history=True)
+    question = {"product": "MLflow"}
+    answer = {"product": "MLflow", "text": TEST_CONTENT}
+    with _mock_request(return_value=_mock_chat_completion_response()):
+        model = create_openai_llmchain()
+        with mlflow.start_run() as run:
+            model.invoke(question)
+        loaded_table = mlflow.load_table("inference_history.json", run_ids=[run.info.run_id])
+        loaded_dict = loaded_table.to_dict("records")
+        assert loaded_dict == [{"input": question, "output": answer}]
+
+        # inference history is appended to the same table
+        with mlflow.start_run(run.info.run_id):
+            model.invoke(question)
+        model.invoke(question)
+        loaded_table = mlflow.load_table("inference_history.json", run_ids=[run.info.run_id])
+        loaded_dict = loaded_table.to_dict("records")
+        assert loaded_dict == [{"input": question, "output": answer}] * 3
+
+        with pytest.raises(MlflowException, match="Please end current run when autologging is on "):
```

Comment:
Can we not use the active run? Is this because langchain will create a new child run in this instance of having an active run context? 
It would be great if we could allow users to specify their own run here wherein we use their activated run context for logging and if langchain is starting a new run on behalf of users, we disable that functionality iff a run is already active. The user-facing run context is likely where users will be doing things like setting tags, extra requirements, code paths, etc... we most certainly do not want to prevent them from doing that.

Reply:
Let's use the client API :)